tests/trainer/logging_tests/test_grad_norm_track.py

```diff
@@ -0,0 +1,123 @@
+        xs, ys = batch
+        out = self.forward(xs)
+        return {'loss': self.loss(out, ys)}
+
+
+def test_grad_norm_track_aggregated_over_parameters():
+    model = TestModel()
+    trainer = Trainer(track_grad_norm=2,
+                      log_every_n_steps=1,
+                      fast_dev_run=True)
+
+    trainer.fit(model)
+
+    first_optimizer = ('grad_2.0_norm_first.weight_mean',
+                       'grad_2.0_norm_first.weight_std',
+                       'grad_2.0_norm_first.bias_mean',
+                       'grad_2.0_norm_first.bias_std',
+                       'grad_2.0_norm_total_mean',
+                       'grad_2.0_norm_total_std')
+    assert all([trainer.logged_metrics[key] > 0.0 for key in first_optimizer]), f'Grad norm not logged for first optimizer'
+
+    second_optimizer = ('grad_2.0_norm_second.weight_mean',
+                        'grad_2.0_norm_second.weight_std',
+                        'grad_2.0_norm_second.bias_mean',
+                        'grad_2.0_norm_second.bias_std',
+                        'grad_2.0_norm_total_mean',
+                        'grad_2.0_norm_total_std')
+    assert all([trainer.logged_metrics[key] > 0.0 for key in second_optimizer]), f'Grad norm not logged for second optimizer'
+
+def test_grad_norm_track_aggregated_over_optimizers_and_parameters():
+    model = TestModel()
+    trainer = Trainer(track_grad_norm=2,
+                      track_grad_norm_mode='optimizer+parameters',
+                      log_every_n_steps=1,
+                      fast_dev_run=True)
+
+    trainer.fit(model)
+
+    first_optimizer = ('opt_0_grad_2.0_norm_first.weight_mean',
+                       'opt_0_grad_2.0_norm_first.weight_std',
+                       'opt_0_grad_2.0_norm_first.bias_mean',
+                       'opt_0_grad_2.0_norm_first.bias_std',
+                       'opt_0_grad_2.0_norm_total_mean',
+                       'opt_0_grad_2.0_norm_total_std')
+    assert all([trainer.logged_metrics[key] > 0.0 for key in first_optimizer]), f'Grad norm not logged for first optimizer'
+
+    second_optimizer = ('opt_1_grad_2.0_norm_second.weight_mean',
+                        'opt_1_grad_2.0_norm_second.weight_std',
+                        'opt_1_grad_2.0_norm_second.bias_mean',
+                        'opt_1_grad_2.0_norm_second.bias_std',
+                        'opt_1_grad_2.0_norm_total_mean',
+                        'opt_1_grad_2.0_norm_total_std')
+    assert all([trainer.logged_metrics[key] > 0.0 for key in second_optimizer]), f'Grad norm not logged for second optimizer'
+
+def test_grad_norm_track_aggregated_over_optimizers():
+    model = TestModel()
+    trainer = Trainer(track_grad_norm=2,
+                      track_grad_norm_mode='optimizer',
+                      log_every_n_steps=1,
+                      fast_dev_run=True)
+
+    trainer.fit(model)
+
+    first_optimizer = ('opt_0_grad_2.0_norm_mean',
```

Comment:
in Optimizer mode, norm and norm_total should be the same. Duplicated information

Reply:
And not sure if `std` makes sense.

Reply:
> in Optimizer mode, norm and norm_total should be the same. Duplicated information

Not necessarily, as the code is today, I think. If you have a dense layer with weights and bias and you aggregate over an optimizer you will get `norm_mean = mean(weight.norm, bias.norm)` calculated by grad_norm_tracker. While `norm_total=mean( (weight+bias).norm )`, where the `(weight + bias).norm` is calculated in  GradInformation.py. 

In my opinion, the most natural thing to do would be norm_total.